src/openapi/streaming/streaming.js

```diff
@@ -221,13 +238,22 @@ function onConnectionStateChanged(nextState) {
             break;
 
         case this.CONNECTION_STATE_CONNECTED:
-            log.info(LOG_AREA, 'Connection connected');
-
             this.retryCount = 0;
             // if *we* are reconnecting (as opposed to transport reconnecting, which we do not need to handle specially)
-            if (this.reconnecting || this.isReset) {
+            if (this.reconnecting) {
                 resetSubscriptions.call(this, this.subscriptions);
                 this.reconnecting = false;
+            }
+
+            // During streaming service switching or reconnect, wait for all subscriptions to unsubscribe before resubscribing again
+            // to avoid hitting throttling limit on subscriptions
+            if (this.isReset) {
+                this.allUnsubscribePendingPromise.then(() => {

```

Comment:
I'm wondering if we should instead merge this:
https://github.com/SaxoBank/openapi-clientlib-js/pull/323

This will ensure that if we call reset that (1) we unsubscribe (2) we do not re-subscribe until the unsubscribe is done.

Reply:
Having gone a bit further, I'd rather undo everything you've changed about waiting for unsubscribes and instead call reset after having newly connected.

Reply:
sounds good to me